pkg/clusteragent/admission/mutate/agent_sidecar/agent_sidecar.go

```diff
@@ -57,7 +58,7 @@ type Webhook struct {
 	// These fields store datadog agent config parameters
 	// to avoid calling the config resolution each time the webhook
 	// receives requests because the resolution is CPU expensive.
-	profilesJSON                 string
+	profileOverrides             *[]ProfileOverride
```

Comment:
Question:

Why do we need `*[]ProfileOverride` here? Isn't `[]ProfileOverride` enough?